sql/moz-fx-data-marketing-prod/acoustic/contact_raw_v1/query.sql

```diff
@@ -1,5 +1,19 @@
+CREATE TEMP FUNCTION yes_no_to_int(val STRING) AS (
+  CASE
+    WHEN LOWER(val) = "no"
+      THEN 0
+    WHEN LOWER(val) = "yes"
+      THEN 1
+    ELSE CAST(val AS INTEGER)
```

Comment:
I don't have visibility on the contents of these fields but since they're all type STRING, this would fail for any string different than (yes, no). Shall we add a case for safety?
```suggestion
    WHEN NULL
      CAST(val AS INTEGER)
    ELSE
      CAST(NULL AS INTEGER)
```

Reply:
not sure what `WHEN NULL` would do here, but probably not what we want? I suggest SAFE_CAST which will `null` everything that can't be cast

Reply:
huh? Where did that come from? 🤔

Reply:
Given we've already had unforeseen changes to the data I'd actually recommend _not_ using `SAFE_CAST()` and leaving this logic as is, because if the data changes yet again (e.g. maybe they start sending "true" and "false" strings) we want this to fail so we can potentially update the ETL rather than nulling out anything we're not expecting and effectively losing data.